### DISCUSSION TITLE
Outputs from shazam.recognize_song(file) ?

### DISCUSSION THREAD

**Discussion by MoppyDev** [shazamio/ShazamIO#68] [Q&A]
Is there a way to tell what timestamp in the song the file ended at?

Say I provide a 10 second file to the recognize_song method, using the provided outputs is there a way to determine the file was taken at 1:10 seconds into the song for example?

**Comment by maukaEast** [comment]
This would be fantastic to implement. It looks like it may be applicable using some of the XXXOffset functions in the API - described by developers in https://medium.com/@alex.telek/behind-the-scenes-of-shazams-real-time-lyrics-feature-8feeb628ad. If you could pull these data, figuring out song position doesn't seem too difficult.

EDIT: figured out 
        out = await shazam.recognize_song(title) 
	val_out = out['matches'][0]
	sample_offset = val_out.get('offset')

will give you the general (in seconds) place in the identified song your sample starts. I'm not sure how exact it is, but I ran a few tests and it seems pretty good.
